src/modules/python/pbs/v1/_pmi_utils.py

```diff
@@ -0,0 +1,158 @@
+#  agreement for PBS Pro version 14 or later has been executed in writing with Altair.
+#   
+#  Altair’s dual-license business model allows companies, individuals, and 
+#  organizations to create proprietary derivative works of PBS Pro and distribute 
+#  them - whether embedded or bundled with other software - under a commercial 
+#  license agreement.
+#  
+#  Use of Altair’s trademarks, including but not limited to "PBS™", 
+#  "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's 
+#  trademark licensing policies.
+ *
+ */
+"""
+__doc__ = """
+$Id$
+Utility power functions.
+"""
+
+import pbs
+import os
+import sys
+
+
+def _pbs_conf(confvar):
+    # Return the value of a setting in the pbs.conf file if it exists.
+    # Save the values in a global dictionary for future use.
+
+    if confvar in os.environ:
+        return os.environ[confvar]
+
+    global pmi_pbsconf
+    if "pmi_pbsconf" not in globals():
+        pmi_pbsconf = dict()
+        cfile = "PBS_CONF_FILE"
+        if cfile in os.environ:
+            pbsconf = os.environ[cfile]
+        else:
+            pbsconf = "/etc/pbs.conf"
+
+        try:
+            fp = open(pbsconf)
+        except:
+            pbs.logmsg(pbs.DEBUG, "%s: conf file not found" % pbsconf)
+            pass
+        else:
+            for line in fp:
+                line = line.strip()
+                # ignore empty lines or those beginning with '#'
+                if line == "" or line[0] == "#":
+                    continue
+                var, eq, val = line.partition('=')
+                if val == "":
+                    continue
+                pmi_pbsconf[var] = val
+            fp.close()
+    if confvar in pmi_pbsconf:
+        return pmi_pbsconf[confvar]
+    else:
+        return None
+
+
+def _is_node_provisionable():
+    """
+    Check if the local machine is running either the pbs_server or
+    pbs_sched.  If either of these are running, provisioning
+    must not be automatically enabled.
+    """
+    serv = _pbs_conf("PBS_START_SERVER")
+    if serv is not None and serv == "1":
+        return False
+
+    sched = _pbs_conf("PBS_START_SCHED")
+    if sched is not None and sched == "1":
+        return False
+    return True
+
+
+def _get_hosts(job):
+    """
+    Form a list of unique short hostnames from a pbs job.
+    The short names are used even if the hostnames from exec_host2
+    are FQDNs because SGIMC seems to have a bug in that it will
+    not accept FQDNs.
+    """
+    hosts = str(job.exec_host2)
+    pbs_nodes = sorted(set([x.partition(':')[0].partition('.')[0]
+                            for x in hosts.split('+')]))
+    return pbs_nodes
+
+
+def _jobreq(job, name):
+    """
+    Get a requested resource from a job.
+    """
+    val = str(job.schedselect).partition(name + '=')[2]
+    if len(val) == 0:
+        return None
+    else:
+        return val.partition('+')[0].partition(':')[0]
+
+
+def _get_vnodes(job):
```

Comment:
I'd suggest you rename it to _get_vnode_names because essentially that's what you are getting.
It seems confusing when you use _get_vnodes and then use it to get the vnode using _svr_vnode() function